src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,368 @@
+        ],
+        params=[
+            "inflation_rate",
+            "interest_during_construction_nominal",
+            "tax_rate_federal_and_state",
+            "interest_rate_calculated_real",
+            "interest_rate_nominal",
+            "rate_of_return_on_equity_nominal",
+            "rate_of_return_on_equity_calculated_real",
+        ],
+    )
+    scenario_table: TableUnstacker = TableUnstacker(
+        idx=rate_table.idx + ["scenario_atb"],
+        params=[
+            "debt_fraction",
+            "wacc_real",
+            "wacc_nominal",
+            "capital_recovery_factor",
+            "fuel_cost",
+            "fixed_charge_rate",
+        ],
+    )
+    tech_detail_table: TableUnstacker = TableUnstacker(
+        idx=scenario_table.idx
+        + [
+            "technology_description_detail_1",
+            "technology_description_detail_2",
+        ],
+        params=[
+            "capex",
+            "capacity_factor",
+            "opex_fixed",
+            "levelized_cost_of_energy",
+            "opex_variable",
+            # 2023 only params
+            "capex_overnight_additional",
+            "construction_finance_factor",
+            "grid_connection_cost",
+            "heat_rate",
+            "heat_rate_penalty",
+            "net_output_penalty",
+            "capex_overnight",
+        ],
+    )
+
+    @property
+    def params_all(self: Self) -> list[str]:
+        """Compilation of all of the parameter values from each of the tables.
+
+        Also check if there are no duplicate params. We expect all of the parameters
+        across the :class:`TableUnstacker` to be unique.
+        """
+        params_all = (
+            self.rate_table.params
+            + self.scenario_table.params
+            + self.tech_detail_table.params
+        )
+        assert not {x for x in params_all if params_all.count(x) > 1}
+        return params_all
+
+
+def broadcast_fcr_param_across_tech_detail(
+    nrelatb_unstacked: pd.DataFrame, idx_unstacked: list[str]
+) -> pd.DataFrame:
+    """For older years, broadcast the FCR parameter across the technical detail columns.
+
+    We want to table schema to be consistent for all years of ATB data. Mostly the parameters
+    have the same primary keys across all of the years. But the ``fcr`` parameter is the
+    only exception. For the older years (pre-2023), the FCR parameter is not variable
+    based on tech detail so we are going to broadcast the pre-2023 fcr values acorss the
+    tech details that exist in the data.
+    """
+    nrelatb_unstacked_fcr = nrelatb_unstacked.reset_index()
+    mask_fcr = (
+        nrelatb_unstacked_fcr.fcr.notnull()
+        & (nrelatb_unstacked_fcr.report_year < 2023)
+        # there are weirdly some fcr nuclear records that have a bunch of data
+        # bc these records are not sparse, we do not want to broadcast these records
+        # we use the columns of nrelatb_unstacked bc those of the params names
```

Comment:
I know this is still WIP, but just a flag here to not forget to complete this thought!